src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -100,103 +78,120 @@ public override void Accept(OperationVisitor visitor)
             return visitor.VisitInvocationExpression(this, argument);
         }
 
-        internal static ImmutableArray<IArgument> DeriveArguments(ImmutableArray<BoundExpression> boundArguments, ImmutableArray<string> argumentNamesOpt, ImmutableArray<int> argumentsToParametersOpt, ImmutableArray<RefKind> argumentRefKindsOpt, ImmutableArray<Symbols.ParameterSymbol> parameters, SyntaxNode invocationSyntax)
+        internal static ImmutableArray<IArgument> DeriveArgumentsInEvaluationOrder(ImmutableArray<BoundExpression> boundArguments, ImmutableArray<string> argumentNames, ImmutableArray<int> argumentsToParameters, ImmutableArray<RefKind> argumentRefKinds, ImmutableArray<Symbols.ParameterSymbol> parameters, SyntaxNode invocationSyntax)
         {
-            ArrayBuilder<IArgument> arguments = ArrayBuilder<IArgument>.GetInstance(boundArguments.Length);
-            for (int parameterIndex = 0; parameterIndex < parameters.Length; parameterIndex++)
+            PooledHashSet<int> matchedParameters = PooledHashSet<int>.GetInstance();
+            ArrayBuilder<IArgument> evaluationOrderArguments = ArrayBuilder<IArgument>.GetInstance(parameters.Length);
+            for (int argumentIndex = 0; argumentIndex < boundArguments.Length; argumentIndex++)
             {
-                int argumentIndex = -1;
-                if (argumentsToParametersOpt.IsDefault)
+                int parameterIndex = argumentsToParameters.IsDefault ? argumentIndex : argumentsToParameters[argumentIndex];
+                IArgument argument = DeriveArgument(parameterIndex, argumentIndex, boundArguments, argumentNames, argumentRefKinds, parameters, invocationSyntax);
+                evaluationOrderArguments.Add(argument);
+                matchedParameters.Add(parameterIndex);
```

Comment:
>matchedParameters.Add(parameterIndex); [](start = 16, length = 38)

Is it possible to run into the same parameter twice? For example, a combination of a positional argument with named argument, or a combination of two named arguments.